### PR TITLE
Gauntlet !pb

### PR DESCRIPTION
**Pull request by mfboulos**
Addresses !pb for the gauntlet, as requested in #9459.

I modified existing PB and kill duration patterns, I don't know of any false positives that would incur so feel free to strike that one down if necessary.

I haven't actually tested this because the only account I have that's able to do so is a hardcore, and I don't want to risk the client crashing or anything like that while I'm in the gauntlet :)

### CODE REVIEW

**Comment by Bram91** [comment]
Just tested it in the regular gauntlet, working as expected for that. Wont be able to try corrupted since I still struggle with that myself.

**Comment by Adam-** [comment]
This should probably include tests too. Does anyone have a screenshot of the completion messages?

**Comment by SebastiaanVanspauwen** [comment]
Regular gauntlet:
image
Corrupted gauntlet:
image

**Comment by Adam-** [comment]
I'm not sure if it is safe to assume eg `Challenge duration: 10:24. Personal best: 7:59` is always the Gauntlet like this is doing?

**Comment by mfboulos** [comment]
The Gauntlet follows the same sort of patterns as the kill duration and PB patterns. I asked a user to help me with the corrupted gauntlet completion pattern after a completion and he sent me this:

image

Which is why I handled it by generalizing those patterns to better suit how they're used (if and only if `lastBossKill` is non-null) and simply using the variation in patterns between normal and corrupt gauntlets to populate `lastBossKill` with the respective string name.

**Comment by Adam-** [comment]
What about https://github.com/runelite/runelite/compare/master...Adam-:gauntpb ? It is slightly less elegant but should be more robust.

File: runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java [hunk 1]
```diff
@@ -290,6 +291,13 @@ else if (result.equals("were defeated"))
 
 			setKc("Barrows Chests", kc);
 		}
+		
+		matcher = GAUNTLET_PATTERN.matcher(message);
+		// TODO: set lastBossKill in kc command instead
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: what is this referring to?

Comment (mfboulos) [reply runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java]: I had a hunch that the messages were being sent as one, and that I could do that. Meant the kc message, but either way it's not relevant anymore. Must have left that in on accident, apologies.

**Comment by Adam-** [comment]
The reason this doesn't work is the same issue as #8454 where it doesn't work with gg, since they also send an out of order pb message. So I've fixed that and then fixed this in e27459f1f0a8fd06ba685baea258bd210d5e21fa

**Pull request closed without merging**